ocamldebug.el

```diff
@@ -717,8 +718,48 @@ Obeying it means displaying in another window the specified file and line."
 
 ;;; Miscellaneous.
 
+(defun ocamldebug--read-from-file (file)
+  "Read FILE as a list of sexps. If invalid syntax, return nil and message the error."
+  (with-temp-buffer
+    (save-excursion
+      (insert "(\n")
+      (insert-file-contents file)
+      (goto-char (point-max))
+      (insert "\n)\n"))
+    (condition-case err
+        (read (current-buffer))
+      ((error err) (progn (message "Error reading file %S: %S" file err) nil)))))
+
```

Comment:
The above `condition-case` could be replaced with `with-demoted-errors`.

Reply:
[ It's worse than not very nice, because hysterical raisins make this syntax unsupported for now. ]
I was assuming that such errors should virtually never happen and would be considered as bugs that need to be fixed, in which case the relevant info is in the backtrace already.
Anyway, I'm OK with leaving it as it is, thanks.